### ISSUE TITLE
php 8.1 or 8.2

### ISSUE THREAD

**Issue by BoneHead514** [DanWin/le-chat-php#143]
is there a way to make your chat work with php 8.1 or 8.2  as you have to install all the old stuff and it levies the server open to hackers

**Comment by DanWin** [comment]
Hello @BoneHead514 , it should work just fine with the latest PHP versions. In fact, it also works with PHP 8.3

**Comment by rexzooly** [comment]
Last time I tested this was on 8.1 and it worked fine, @BoneHead514 why do you believe this don't work.

**Comment by BoneHead514** [comment]
i install your hosting server and then install the chat on the host like you use to have Daniel but it wont work it keeps looping to the SA sign in screen

**Comment by DanWin** [comment]
@BoneHead514 that sounds like you may have an issue with SQL statements not being run? Check the error logs of your hosting. There should be some warnings or errors logged in that case.

**Comment by ThePrivate000** [comment]
Hey @DanWin, is there any way you can update the chat? because when i setup the php8.2 and 8.3, it shows like a white page when i have my site up running, and doesn't show the contents of the php versions. Pls update the chat file, or code

**Comment by DanWin** [comment]
Hello @ThePrivate000 , that sounds like a syntax error. Check your error logs. If you need help resolving the issue, post the error here. The latest PHP versions are supported.

**Comment by ThePrivate000** [comment]
@DanWin yes i'll send you the error issue of php, i'd used this new release of php, the command here is sudo apt install php8.2-fpm php8.2-common php8.2-curl php8.2-intl php8.2-mbstring php8.2-xmlrpc php8.2-mysql php8.2-gd php8.2-pgsql php8.2-xml php8.2-cli php8.2-imagick php8.2-bcmath php8.2-gmp

but that php command couldn't find any package on my debian 10 linux: >>>

  E: Couldn't find any package by regex 'php8.2-gmp'

and i did this in all linux Operating systems and non of them worked with that new realease,

**Comment by ThePrivate000** [comment]
It;s possible that your chat file code needs an upgrade so it can work an all versions within php because at the moment im now stress with this php crap and the chat code not having it to work.

**Comment by DanWin** [comment]
@ThePrivate000 This is not an issue with the chat, but rather with installing PHP.
If you are using Debian 10 (buster), then you are likely not able to install php8.2 because it is a very old distribution version that ships with PHP 7.3, which is already end of life since 2021. Consider upgrading to Debian 12 (bookworm) for php8.2. Alternatively if you must stay on this old system, use the php packages from https://deb.sury.org/

**Comment by ThePrivate000** [comment]
@Danwin can you correct me which one is the right php command to put on the terminal in whonix or other linux distrobutions? I've tried the new Debian 12 before, after a friend recommanded me to go with debian 10 instead of the debian 12 linux bookworm, but if i go with that one the debian 12 older new realease, and put the command, whatever php command, will it work on there ?

**Comment by DanWin** [comment]
Using a several years old version is never a good idea, unless you specifically need to support an outdated piece of software that is not compatible with a more modern version. Or would you go and buy a 10 year old phone instead of the latest model?
As for the distribution you are using, each distribution has different package managers and repositories, so I cannot provide you a specific command without first doing some research. Whonix is based on Kicksecure, which is based on Debian, so you can enable Debian repositories to install additional software. See https://www.kicksecure.com/wiki/Install_Software

**Comment by DanWin** [comment]
Hi we waiting for you to reply on Issue section in github

What we need to install for the chat to make its work its not clear

can you do a list in debian of everything who we need to install ? thank you

in the past installing le chat with php 7.4 working well fast now we need to
install this or that or this or that .. seriously .. 

Do a list please thank you  for your friend from tor !

**Comment by DanWin** [comment]
The requirements have not significantly changed in recent years. Just make sure your server meets the requirements outlined in the Readme: https://github.com/DanWin/le-chat-php#installation-instructions
